### ISSUE TITLE
Back-office and images without querystrings return 404

### ISSUE THREAD

**Issue by TWJH** [JimBobSquarePants/UmbracoFileSystemProviders.Azure#142]
Hi,

I've been trying for days to get this working and trawled through previous issues but I still cannot resolve my problem.

I have moved my media content to azure blob storage and manually configured my umbraco app as per the readme. The problem is that only images that have querystrings and are processed by ImageProcessor are rendering. Images in the back-office are not showing and images without querystrings are coming back with a 404 but definitely exist in the blob storage container. 

If I set `interceptAllRequests="true"` in **processing.config** (I know this is not recommended), all the images on the site work but the back-office still doesn't.

This is the **web.config** which is in the /media/ folder in my solution:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<configuration>
  <system.webServer>
    <handlers>
      <clear />
      <add name="StaticFileHandler" path="*" verb="*" preCondition="integratedMode" type="System.Web.StaticFileHandler" />
      <add name="StaticFile" path="*" verb="*" modules="StaticFileModule,DefaultDocumentModule,DirectoryListingModule" resourceType="Either" requireAccess="Read" />
    </handlers>
  </system.webServer>
</configuration>
```

This is my **security.config**

```xml
<?xml version="1.0" encoding="utf-8"?>
<security>
  <services>
    <service name="LocalFileImageService" type="ImageProcessor.Web.Services.LocalFileImageService, ImageProcessor.Web" />
    <service prefix="media/" name="CloudImageService" type="ImageProcessor.Web.Services.CloudImageService, ImageProcessor.Web">
      <settings>
        <setting key="Container" value="*****"/>
        <setting key="MaxBytes" value="52428800"/>
        <setting key="Timeout" value="30000"/>
        <setting key="Host" value="http://*****.blob.core.windows.net/"/>
      </settings>
    </service>
  </services>
</security>
```

This is my **FileSystemProviders.config**

```xml
<?xml version="1.0"?>
<FileSystemProviders>

  <!-- Media -->
  <Provider alias="media" type="Our.Umbraco.FileSystemProviders.Azure.AzureBlobFileSystem, Our.Umbraco.FileSystemProviders.Azure">
    <Parameters>
      <add key="containerName" value="***"/>
      <add key="rootUrl" value="https://***.blob.core.windows.net/"/>
      <add key="connectionString"
        value="DefaultEndpointsProtocol=https;AccountName=***;AccountKey=***"/>
      <add key="maxDays" value="365"/>
      <add key="useDefaultRoute" value="true"/>
      <add key="usePrivateContainer" value="false"/>
    </Parameters>
  </Provider>

</FileSystemProviders>
```

Precompile is not set and the problem is also there when I debug locally.

The versions I have installed are as follows:

UmbracoCms - 7.5.14
UmbracoFileSystemProviders.Azure - 1.0.3
ImageProcessor - 2.7.0.100
ImageProcessor.Web - 4.10.0.100
ImageProcessor.Web.Config - 2.5.0.100

**Comment by JimBobSquarePants** [comment]
Reads to me like you haven't got the Static File Handler enabled in the media folder.

https://github.com/JimBobSquarePants/UmbracoFileSystemProviders.Azure#virtual-path-provider

I would double check all your config files against the readme. You definitely do not want intercept all request enabled.

**Comment by TWJH** [comment]
Thanks for getting back to me.

I have this in my root web.config:

```xml
  <location path="Media">
    <system.webServer>
      <handlers>
        <remove name="StaticFileHandler" />
        <add name="StaticFileHandler" path="*" verb="*" preCondition="integratedMode" type="System.Web.StaticFileHandler" />
      </handlers>
    </system.webServer>
  </location>
```

I have posted my /media/ web.config in my previous post.

I can honestly say I have spent hours and hours checking and re-checking and going over my configuration as per the readme. 

I have been handed this project from previous developers and my job has been to refactor it from the rats nest it was to something a bit more civilised to work with.

I initially tried to install this package via the umbraco package but it failed every time with an unspecific error saying something like "an exception occurred" with no details and nothing in the logs to provide more details which is why the manual config. Could this be indicative of something else being awry in this project which may be causing the issues I am seeing?
